vmm/src/config.rs

```diff
@@ -131,25 +131,43 @@ fn parse_iommu(iommu: &str) -> Result<bool> {
 }
 
 #[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct CpusConfig(pub u8);
+pub struct CpusConfig{
+    pub cpu_count: u8,
+}
 
 impl CpusConfig {
     pub fn parse(cpus: &str) -> Result<Self> {
-        Ok(CpusConfig(
-            cpus.parse::<u8>().map_err(Error::ParseCpusParams)?,
-        ))
+        // Split the parameters based on the comma delimiter
+        let params_list: Vec<&str> = cpus.split(',').collect();
+
+        let mut cpu_count_str: &str = "";
+
+        for param in params_list.iter() {
+            if param.starts_with("cpu_count=") {
```

Comment:
Here you're modifying the expectation about the CLI too. This needs further changes to the documentation if we accept this change.

But I'm wondering why you couldn't leave the CpusConfig structure the way it was? Was the deserialization failing?

Reply:
@sboeuf yeah , was a deserialization  error when http request tried to get information from vmconfig.
As defined in the api yaml, I am sending :
`{"cpus":{"cpu_count":5},"memory":{"size":0},"kernel":{"path":""},"cmdline":{"args":"hola"},"rng":{"src":""},"serial":{"mode":"Off"},"console":{"mode":"Off"}}`

but it fails saying `SerdeJsonDeserialize(Error("invalid type: map, expected u8", line: 1, column: 8))` .